dns_cert_checker.py

```diff
@@ -43,16 +42,45 @@
 from sslyze.scanner.server_scan_request import ServerScanResult
 from sslyze.server_connectivity import ServerConnectivityTester
 from sslyze.server_setting import ServerNetworkLocationViaDirectConnection
+import time
+from typing import Dict, Iterable, List, Optional, Set, Tuple, Union
 
 RUN_TIME_TIMESTAMP = int(time.time())
 
 logging.basicConfig()
 LOGGER = logging.getLogger()
 
 
+def _emit_stats(
+    endpoint: str,
+    metric: str,
+    fields: Dict[str, float],
+    tags: Dict[str, str]
+):
+    """
+    Emit stats in influx format to a UDP endpoint.
+
+    :param endpoint: A string with a format of <dns endpoint>:<port>
+    :param metric: the name of the metric to produce
+    :param fields: a dictionary of fields and their values
+    :param tags: a dictionary of tags and their values
+    """
+    endpoint, port = endpoint.split(":")
+    tag_str = ",".join([f"{k}={v}" for k, v in tags.items()])
+    field_str = ",".join([f"{k}={v}" for k, v in fields.items()])
+    batch = f"{metric},{tag_str} {field_str}"
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.sendto(batch.encode("utf-8"), (endpoint, int(port)))
+    except Exception as e:
+        LOGGER.error(f"Failed to send ({batch}) to {endpoint}:{port} :: {e}")
+
+
 def parse_dns_dict(
     dns_export: Dict[str, List[Dict]],
-    dns_resolver: Optional[dns.resolver.Resolver] = None
+    zone_name: str,
+    dns_resolver: Optional[dns.resolver.Resolver] = None,
+    name_filters: Optional[List] = [],
```

Comment:
should initialize as None, then check in the function to declare it as an empty list if it is None